calculator/script.js

```diff
@@ -0,0 +1,232 @@
+const calculatorButtonsHolder = document.querySelector(".calculator__buttons-holder");
+const calculatorShowAll = document.querySelector(".calculator__show-all");
+const calculatorShow = document.querySelector(".calculator__show");
+let number1 = '';
+let number2 = '';
+let active = 'number1';
+let currentResult = 0;
+let sign = '';
+const disabledButtons = document.querySelectorAll('.disabled');
+let previousOpperation = 'resultOperation';
+
+
+const buttonOnClick = (e)=>{
+    if(e.target.classList.contains("button")){
+        if(previousOpperation === "cleenOperation"){
+            calculatorShow.textContent = "";
+            calculatorShowAll.textContent = "";
+        }
+        if(calculatorShow.textContent === "0"){
+            calculatorShow.textContent = "";
+        }
+        if(e.target.classList.contains("number")){
+            numberPress(e.target.textContent);
+        } else if(e.target.classList.contains("simple-operation")){
+            simpleOperationPressed(e.target.textContent);
+        } else if(e.target.classList.contains("square-operation")){
+            squareOperationPressed(e.target.textContent);
+        }else if(e.target.classList.contains("sign-change")){
+            signChangePressed(e.target.textContent);
+        }else if(e.target.classList.contains("decimal-sign")){
+            decimalSignPressed(e.target.textContent);
+        }else if(e.target.classList.contains("cleen")){
+            cleenOperationPressed((e.target.textContent));
+        }else if(e.target.classList.contains("result")){
+            resultOperationPressed(e.target.textContent);
+        }
+    }
+}
+
+//Numper press
+const numberPress = (number) => {
+    document.querySelector('.disabled-cleen').removeAttribute("disabled", "disabled");
+    let regexp = /\./g;
+    if (previousOpperation === 'resultOperation'){
+        currentResult = 0;
+        calculatorShow.textContent = "";
+            calculatorShowAll.textContent = "";
+
+    }else if ((previousOpperation !== 'number') && (previousOpperation !=='cleenLastOperation')){
+        calculatorShow.textContent = ' ';
+    }
+    for (let i=0; i<disabledButtons.length; i++){
+        disabledButtons[i].removeAttribute("disabled", "disabled");
```

Comment:
I guess it is not working as expected. Because you initialize disabledButtons only once on page init. And then you toggle this class in different elements.